apps/passport-client/src/dispatch.ts

```diff
@@ -942,3 +956,67 @@ async function promptToAgreePrivacyNotice(state: AppState, update: ZuUpdate) {
     });
   }
 }
+
+/**
+ * Merge in a PCD collection from an import of backed-up data.
+ */
+async function mergeImport(
+  state: AppState,
+  update: ZuUpdate,
+  collection: PCDCollection,
+  pcdsToMergeIds: Set<PCD["id"]>
+) {
+  const userHasExistingSemaphoreIdentityPCD =
+    state.pcds.getPCDsByType(SemaphoreIdentityPCDTypeName).length > 0;
+
+  const userHasExistingEmailPCD =
+    state.pcds.getPCDsByType(EmailPCDTypeName).length > 0;
+
+  const pcdCountBeforeMerge = state.pcds.getAll().length;
+
+  const filterFunction = (pcd: PCD, target: PCDCollection): boolean => {
+    return (
+      pcdsToMergeIds.has(pcd.id) &&
+      !(pcd.type === EmailPCDTypeName && userHasExistingEmailPCD) &&
+      !(isSemaphoreIdentityPCD(pcd) && userHasExistingSemaphoreIdentityPCD) &&
+      !(
+        isSemaphoreIdentityPCD(pcd) &&
+        pcd.claim.identity.getCommitment().toString() !== state.self.commitment
+      ) &&
+      !target.hasPCDWithId(pcd.id)
+    );
+  };
+
+  const packages = await getPackages();
+
+  // This is a clunky way to clone an object.
```

Comment:
Good job noticing this.  I'm nervous about race-conditions, which aren't ever as infinitessimal as you want them to be.
If there's a background sync happening at this moment, the results here could clobber that.

In future, we need some sort of synchronization of PCD mutations to ensure no races on read-modify-write.

For now, I'd settle for a shallow clone which copies the PCD array and folder map but leaves the PCD contents alone.  In practice, we never mutate PCD contents anyway.

Reply:
This is another good place to explicitly document that the solution to this problem is relying on the import screen not enabling background sync.